internal/contracts/tests/common_test.go

```diff
@@ -36,7 +36,7 @@ var (
 	}
 )
 
-func TestMetadataInsertionAndRetrieval(t *testing.T) {
+func TestCOMMON04MetadataInsertionAndRetrieval(t *testing.T) {
```

Comment:
isn't it missing the "enabled or not" part? maybe would be a good idea to have the statements on a comment above the test to be easy to know what the test should be covering, without going back to the main document